<launch>
  <!-- Arguments -->
  <arg name="robot_name" default="robot_0" />
  <arg name="cmd_vel_topic" default="$(arg robot_name)/cmd_vel" />
  <arg name="odom_topic" default="$(arg robot_name)/odom" />
  <arg name="move_forward_only" default="false"/>

  <!-- move_base -->
  <group ns="$(arg robot_name)">
  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    <param name="base_local_planner" value="dwa_local_planner/DWAPlannerROS" />

    <rosparam file="$(find physarum_navigation)/param/costmap_common_params_caminhao.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find physarum_navigation)/param/costmap_common_params_caminhao.yaml" command="load" ns="local_costmap" />


    <param name="global_costmap/scan/sensor_frame" value="$(arg robot_name)/base_laser_link" />
    <param name="global_costmap/global_frame" value="map" />
    <param name="global_costmap/static_layer/topic" value="map"/>
    <param name="global_costmap/obstacle_layer/laser_scan_sensor/topic" value="/$(arg robot_name)/base_scan"/>
    <param name="global_costmap/obstacle_layer/laser_scan_sensor/sensor_frame" value="$(arg robot_name)/base_laser_link"/>


    <rosparam file="$(find physarum_navigation)/param/local_costmap_params.yaml" command="load" />
    <param name="local_costmap/global_frame" value="$(arg robot_name)/odom" />
    <param name="local_costmap/robot_base_frame" value="$(arg robot_name)/base_footprint" />
    <param name="local_costmap/obstacle_layer/laser_scan_sensor/topic" value="/$(arg robot_name)/base_scan"/>
    <param name="local_costmap/obstacle_layer/laser_scan_sensor/sensor_frame" value="$(arg robot_name)/base_laser_link"/>



    <rosparam file="$(find physarum_navigation)/param/global_costmap_params.yaml" command="load" />
    <param name="global_costmap/robot_base_frame" value="$(arg robot_name)/base_footprint" />
    <rosparam file="$(find physarum_navigation)/param/move_base_params.yaml" command="load" />
    <rosparam file="$(find physarum_navigation)/param/dwa_local_planner_params_caminhao.yaml" command="load" />
    
    <param name="DWAPlannerROS/min_vel_x" value="0.0" if="$(arg move_forward_only)" />
  </node>
   </group>
</launch>